test/common.js

```diff
@@ -45,3 +59,39 @@ export function expectAchievementIdsNested(promise, name,
 			.property(`${name}.titleId`, titleId)
 	]);
 }
+
+export function generateProcessEdit(func, name, rev) {
+	return () => testData.createEditor()
+		.then((editor) => Achievement.processEdit(orm, editor.id))
+		.then((edit) => edit[func][`${name} ${rev}`]);
+}
+
+export function generateProcessEditNamed(func, name) {
+	return () => new Editor({name: testData.editorAttribs.name})
+		.fetch()
+		.then((editor) =>
+			Achievement.processEdit(orm, editor.id)
+		)
+		.then((edit) => edit[func][name]);
+}
+
+export function rewireAchievement(rewiring) {
+	return () => Achievement.__set__(rewiring);
+}
+
+export function rewireTypeCreation(name, threshold) {
+	return () => rewireAchievement({
+		getTypeCreation:
+			testData.typeCreationHelper(
+				`${name}_revision`, threshold
+			)
+	});
+}
+
+export function testAchievement(rewiring, generator, expectations) {
+	return () => {
```

Comment:
This probably shouldn't return a function. I'd expect it to just be:

    testAchievement(...) {
        rewiring();
        const promise = generator();
        expectations(promise);
    }

Reply:
> So, we'd like you to write a function that takes those as parameters, and either
> a) returns a function that'll test a particular achievement
> b) test the achievement itself

Reply:
OK, so if you want to take that approach, you'll need to do something like:

    const test1 = testAchievement(a, b, c); // at this point test1 is a function taking no parameters generated by testAchievement
    it('...', test1);

Reply:
Shall I disassemble the parameters in the same way?